lrc/Pos/Lrc/Fts.hs

```diff
@@ -0,0 +1,52 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | Base part of  /follow-the-satoshi/ procedure.
+
+module Pos.Lrc.Fts
+       ( followTheSatoshiM
+       ) where
+
+import           Data.List.NonEmpty (fromList)
+import           Universum
+
+import           Pos.Core.Coin      (coinToInteger, unsafeAddCoin)
+import           Pos.Core.Constants (epochSlots)
+import           Pos.Core.Types     (Coin, SharedSeed (..), StakeholderId, mkCoin)
+import           Pos.Crypto         (deterministic, randomNumber)
+import           Pos.Util.Iterator  (MonadIterator (..))
+
+-- | A version of 'followTheSatoshi' that uses an iterator over 'TxOut's
+-- instead of 'Utxo'.
+followTheSatoshiM
+    :: forall m . MonadIterator (StakeholderId, Coin) m
+    => SharedSeed -> Coin -> m (NonEmpty StakeholderId)
+followTheSatoshiM _ totalCoins
+    | totalCoins == mkCoin 0 = panic "followTheSatoshiM: nobody has any stake"
+followTheSatoshiM (SharedSeed seed) totalCoins = do
+    res <- findLeaders (sortOn fst $ zip coinIndices [1..]) (mkCoin 0)
```

Comment:
@pva701 moved this function from `cardano-sl` to `cardano-sl-lrc`. I suppose that's all you need to know within this PR 😉